test/common/tcp_proxy/tcp_proxy_test.cc

```diff
@@ -1927,6 +1927,23 @@ TEST_F(TcpProxyRoutingTest, DEPRECATED_FEATURE_TEST(ApplicationProtocols)) {
   filter_->onNewConnection();
 }
 
+TEST_F(TcpProxyRoutingTest, DEPRECATED_FEATURE_TEST(ClusterNameSet)) {
```

Comment:
is this a deprecated feature test?

Reply:
Not sure why all other tests for TcpProxyRoutingTest are marked DEPRECATED_FEATURE_TEST, hence I marked my test as the same... 
will remove it

Reply:
Removed

Reply:
This is needed, because TcpProxyRoutingTest is using deprecated_v1 config.
@mattklein123 : what's the plan for such classes? should we add new tests somewhere else?

Reply:
Other then that,  PR is ready for your review.. PTAL

Reply:
Please make a new test to test this feature which doesn't use deprecated config, then you can remove the DEPRECATED_FEATURE_TEST

Reply:
Fixed.. PTAL